pkg/controller/plugins.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
```

Comment:
Well... this one will be different because we'll do it right. :)

More seriously, I'm very focused lately on getting the package dependency
tree into a sane state. This means separating interfaces like this so that
plugin code doesn't need to import manager code and vice versa. The
interfaces need to be in common packages, but no actual code should be.

On Mon, Sep 14, 2015 at 7:55 PM, Mark Turansky <notifications@github.com>
wrote:

> In pkg/controller/plugins.go
> <https://github.com/kubernetes/kubernetes/pull/6520#discussion_r39470075>:
>
> > +Copyright 2015 The Kubernetes Authors All rights reserved.
> > +
> > +Licensed under the Apache License, Version 2.0 (the "License");
> > +you may not use this file except in compliance with the License.
> > +You may obtain a copy of the License at
> > +
> > +    http://www.apache.org/licenses/LICENSE-2.0
> > +
> > +Unless required by applicable law or agreed to in writing, software
> > +distributed under the License is distributed on an "AS IS" BASIS,
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package controller
>
> Cloudprovider, Network, and Volume all have their plugins.go and
> interfaces in the top level of the pkg. Why would this one be different?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/6520/files#r39470075>.
>